### PR TITLE
[SDK] Term and MultiTerm TVL

### PR DESCRIPTION
**Pull request by cashd**
- Term TVL (does not account for accrued interest yet)
- MultiTerm TVL, batch version of term tvl
- Added bignumber.js package with a proof of concept impl
- added some constants and utils (isPT, isPY for filtering)

### CODE REVIEW

File: packages/core-v2-sdk/package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
   "dependencies": {
     "@elementfi/base": "*",
     "@elementfi/core-v2-typechain": "*",
+    "bignumber.js": "^9.1.0",
     "evm-bn": "^1.1.2",
```
Comment (ryangoree) [inline packages/core-v2-sdk/package.json:51]: Do we still need `evm-bn` now that we have `bignumber.js`?

File: packages/core-v2-sdk/src/models/Term.ts [hunk 2]
```diff
@@ -28,9 +28,15 @@ export class Term {
...
-    return "0";
+  /**
+   * Gets the TVL of this term, in terms of the underlying token
+   * @todo Does not account for accrued interest, need access to _underlying(ShareState.locked)
+   * @param {number} termId - the term id (expiry)
+   * @return {Promise<string>} total supply represented as a string as a decimal number
+   */
+  async getTVL(): Promise<string> {
+    const balance = await this.multiTerm.dataSource.getTotalSupply(this.id);
+    return balance;
```
Comment (ryangoree) [inline packages/core-v2-sdk/src/models/Term.ts:39]: We can remove the `async` here and just return the promise.

File: packages/core-v2-sdk/src/utils/token/isPT.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { BigNumber } from "ethers";
+
+/**
+ * Determines whether or not the specified tokenId is a principle token.
+ * @param {BigNumber} tokenId - token id
+ * @return {boolean} whether or not is a principle token
+ */
+export function isPT(tokenId: BigNumber): boolean {
+  return !tokenId.toHexString().startsWith("0x8");
+}
```
Comment (ryangoree) [inline packages/core-v2-sdk/src/utils/token/isPT.ts:10]: Nice

File: packages/core-v2-sdk/src/utils/token/isYT.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    // (whereas principal tokens are shorter in length)
+    idHex.length === 66 &&
+    // yield token ids always start with 0x8 (1 in binary)
+    idHex.startsWith("0x8") &&
+    // This is a special yield token used for accounting in the
+    // pools, disregard it.
+    idHex !== UNLOCKED_YT_ID;
+
+  return isYT;
+}
```
Comment (ryangoree) [inline packages/core-v2-sdk/src/utils/token/isYT.ts:23]: Nice

**Review by ryangoree — APPROVED** [review]

File: packages/core-v2-sdk/package.json (hunk 1 above)
Comment (cashd) [reply packages/core-v2-sdk/package.json:51]: evm-bn is just two util functions that makes it easy to go from BigNumber <-> string fixed point number, so when we read/write to the contracts. but when we are simulating logic (trades, apy, etc) we can use this library for higher precision arithmetic.

**Pull request merged**